client/src/modules/Teams/hooks/useDistribution/useDistribution.ts

```diff
@@ -0,0 +1,28 @@
+import { message } from 'antd';
+import { TeamDistributionApi, TeamDistributionDetailedDto } from 'api';
+import { useState, useMemo } from 'react';
+
+export function useDistribution(teamDistributionDetailed: TeamDistributionDetailedDto, courseId: number) {
+  const [distribution, setDistribution] = useState(teamDistributionDetailed);
+  const [loading, setLoading] = useState(false);
```

Comment:
You can use `useLoading` hook instead of manual loading managing

Reply:
fixed